cmd/stack/proxy.go

```diff
@@ -63,6 +66,43 @@ func (c *StackProxyController) GetStore() *StackProxyStore {
 	return c.store
 }
 
+// corsMiddleware creates a CORS middleware that handles preflight requests and adds CORS headers
+func corsMiddleware(allowedOrigins []string, next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		origin := r.Header.Get("Origin")
+
+		// Check if origin is allowed
+		originAllowed := false
+		for _, allowedOrigin := range allowedOrigins {
+			if allowedOrigin == "*" || allowedOrigin == origin {
+				originAllowed = true
+				break
+			}
+		}
+
+		if originAllowed {
+			if origin != "" {
+				w.Header().Set("Access-Control-Allow-Origin", origin)
+			} else if len(allowedOrigins) == 1 && allowedOrigins[0] == "*" {
+				w.Header().Set("Access-Control-Allow-Origin", "*")
+			}
+		}
```

Comment:
We could extract the origin string in the first for loop to reduce if statements and simplify the code.

Reply:
Yes, as long as it is allowed, the code can directly set the header Access-Control-Allow-Origin to the incomling value